File: literary_life_app/lib/models/inspiration.dart
class Inspiration {
  final int id;
  final int userId;
  final int? cycleId;
  final DateTime eventTime;
  final String location;
  final String objectOrEvent;
  final String detailText;
  final String feeling;
  final String keywords;
  final DateTime createdAt;

  Inspiration({
    required this.id,
    required this.userId,
    this.cycleId,
    required this.eventTime,
    this.location = '',
    this.objectOrEvent = '',
    this.detailText = '',
    this.feeling = '',
    this.keywords = '',
    required this.createdAt,
  });

  factory Inspiration.fromJson(Map<String, dynamic> json) {
    return Inspiration(
      id: json['id'],
      userId: json['user_id'],
      cycleId: json['cycle_id'],
      eventTime: DateTime.parse(json['event_time']),
      location: json['location'] ?? '',
      objectOrEvent: json['object_or_event'] ?? '',
      detailText: json['detail_text'] ?? '',
      feeling: json['feeling'] ?? '',
      keywords: json['keywords'] ?? '',
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toCreateJson({int? cycleId}) {
    return {
      if (cycleId != null) 'cycle_id': cycleId,
      'event_time': eventTime.toIso8601String(),
      'location': location,
      'object_or_event': objectOrEvent,
      'detail_text': detailText,
      'feeling': feeling,
      'keywords': keywords,
    };
  }
}

File: literary_life_backend/app/schemas/inspiration.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class InspirationCreate(BaseModel):
    cycle_id: Optional[int] = None
    event_time: Optional[datetime] = None
    location: Optional[str] = ""
    object_or_event: Optional[str] = ""
    detail_text: Optional[str] = ""
    feeling: Optional[str] = ""
    keywords: Optional[str] = ""


class InspirationUpdate(BaseModel):
    location: Optional[str] = None
    object_or_event: Optional[str] = None
    detail_text: Optional[str] = None
    feeling: Optional[str] = None
    keywords: Optional[str] = None


class InspirationResponse(BaseModel):
    id: int
    user_id: int
    cycle_id: Optional[int]
    event_time: datetime
    location: str
    object_or_event: str
    detail_text: str
    feeling: str
    keywords: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
